app/src/main/java/com/leoart/koreanphrasebook/data/repository/DialogsRepository.kt

```diff
@@ -5,68 +5,105 @@ import android.util.Log
 import com.leoart.koreanphrasebook.data.network.firebase.DialogsRequest
 import com.leoart.koreanphrasebook.data.network.firebase.dialogs.models.DialogResponse
 import com.leoart.koreanphrasebook.data.network.firebase.dialogs.models.Replic
+import com.leoart.koreanphrasebook.data.repository.models.EChapter
 import com.leoart.koreanphrasebook.data.repository.models.EDialog
-import io.reactivex.BackpressureStrategy
-import io.reactivex.Flowable
-import io.reactivex.Observable
+import com.leoart.koreanphrasebook.data.repository.models.EPhrase
+import com.leoart.koreanphrasebook.ui.sync.SyncModel
+import com.leoart.koreanphrasebook.utils.toCompletable
+import io.reactivex.*
 import io.reactivex.schedulers.Schedulers
 
 /**
  * DialogsRepository
  *
  * @author Bogdan Ustyak (bogdan.ustyak@gmail.com)
  */
-class DialogsRepository(private val context: Context) {
+class DialogsRepository(private val context: Context) : CachedRepository<DialogResponse>, RefreshableRepository {
 
     companion object {
         val TAG = "DialogsRepository"
     }
 
-    fun getDialogs(): Flowable<List<DialogResponse>> {
+    override fun getItems(): Flowable<List<DialogResponse>> {
         Log.d(TAG, "getDialogs")
         return getDataFromDB()
-                .flatMap {  Flowable.just(mapDialogs(it)) }
                 .doOnNext {
-                    if(it.isEmpty()){
+                    if (it.isEmpty()) {
                         requestFromNetwork()
                     }
                 }
     }
 
-    private fun getDataFromDB(): Flowable<List<EDialog>> {
+    override fun getDataFromDB(): Flowable<List<DialogResponse>> {
         Log.d(TAG, "getDataFromDB")
         return AppDataBase.getInstance(context).dialogDao().getAll()
+                .flatMap {
+                    mapDialogs(it)
+                }
     }
 
-    private fun requestFromNetwork() {
+    override fun requestFromNetwork() {
         Log.d(TAG, "requestFromNetwork")
         DialogsRequest().getAllDialogNames()
-                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.io())
```

Comment:
@greenhorn31 why remove subscribeOn ?

Reply:
because there is subscribeOn() in getAllDialogNames